src/Form/WebformCiviCRMSettingsForm.php

```diff
@@ -70,6 +70,10 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $webform = $this->getWebform();
     $handler_collection = $webform->getHandlers('webform_civicrm');
     $values = $form_state->getValues();
+    if (!isset($values['nid']) && !empty($form['form_state'])) {
```

Comment:
We have the form state, we should use it instead of fetching from the form object.

Reply:
I see. the form state somehow has different values. probably the ones from the confirm form style setup.

Reply:
See https://github.com/colemanw/webform_civicrm/pull/212/commits/d45f8518444656da862d5ec8059bb2bf2605ea9e